strawberry/schema/schema_converter.py

```diff
@@ -190,10 +265,15 @@ def from_schema_directive(self, directive: Any) -> GraphQLDirective:
             ],
             is_repeatable=False,
             args=args,
-            description=strawberry_directive.description,
+            description=description,
         )
 
-    def from_field(self, field: StrawberryField) -> GraphQLField:
+    def from_field(
+        self,
+        field: StrawberryField,
+        description_sources: List[DescriptionSource],
```

Comment:
Hmm looking at this now, I'm unsure how this works and probably will be for users too, potentially leading to accidental leakage.

What happens if I have

```python
@strawberry.type(description_sources=[DescriptionSource.TYPE_DOCSTRINGS])
class Query:
    """AAAAA"""

    @strawberry.resolver(description_sources=[DescriptionSource.RESOLVER_DOCSTRINGS])
    def my_resolver(): ...
    """BBBBB"""

schema = strawberry.Schema(
    query=Query,
    config=StrawberryConfig(description_sources=[DescriptionSource.STRAWBERRY_DESCRIPTIONS]),
)
````

what makes it on the schema? I would definitely like clarification on ownership/nesting and test cases too.

Reply:
Gotcha, so inner attributes' `description_sources` override outer attributes'.

Reply:
Could you make this super-duper clear in the description? I don't want users getting confused